extension/controller/backgroundScript.js

```diff
@@ -60,11 +60,21 @@ const messageListener = function(message, sender) {
         case "detectPageLanguage":
           if (!modelFastText) break;
 
+          languageDetection = Object.assign(new LanguageDetection(), message.languageDetection);
+
+          /*
+           * if we don't support this browser's language, we nede to hide the
+           * page action and bail right away
+           */
+          if (!languageDetection.isBrowserSupported()) {
+            browser.pageAction.hide(sender.tab.id);
```

Comment:
We need the tabid in the background script to do that, since as you noticed, those icons are governed by tabs, and the content scripts ar loaded at the end of the page load (see run_at)

Reply:
ok